stdlib/src/os/fstat.mojo

```diff
@@ -46,7 +46,7 @@ fn _constrain_unix():
 
 
 @value
-struct stat_result(Stringable):
+struct stat_result(Stringable, Writable):
```

Comment:
**Suggestion** Split this change to `stat_result` to be `Writable` along with its tests to a separate PR, please.

Reply:
The goal of this commit was to have a faster representation of `path.stat()` which uses `stat_result` underneath, which was already representable but slow, so to me was in the same scope of "Faster repr() logic" https://github.com/msaelices/mojo/commit/950d36adbe5823c5611038669e14c2a9dd00f594

Do you still want me to split those changes into a new PR?

Reply:
I understand the scope of the commit, but in the future, I'd recommend teasing apart the two ideas (applying `Writable` to a struct and making `repr()` faster for a few types).  They're orthogonal things in my mind and could have been two separate PRs.  It's fine for this one though, but something to consider in the future 😄